### PR TITLE
Cosmetic changes in TSCH 

### PR DESCRIPTION
**Pull request by atiselsts**
Should not affect the functionality.

### CODE REVIEW

**Comment by atiselsts** [comment]
About the RNG change - to confirm that it is a cosmetic-only cjhange, I checked assembly code. I used `diff` in order to compare the versions "before" and "after" and verified that they are indeed identical. 

Here are the listing for two different platforms. For `TARGET=z1`:
<pre>000061b6 <tsch_random_byte>:
    61b6:       0b 14           pushm.a #1,     r11     
    61b8:       4b 4f           mov.b   r15,    r11     
    61ba:       3c 40 6d 4e     mov     #20077, r12     ;#0x4e6d
    61be:       3d 40 c6 41     mov     #16838, r13     ;#0x41c6
    61c2:       1e 42 c2 15     mov     &0x15c2,r14     
    61c6:       1f 42 c4 15     mov     &0x15c4,r15     
    61ca:       b0 13 8a 34     calla   #0x0348a        
    61ce:       3e 50 39 30     add     #12345, r14     ;#0x3039
    61d2:       0f 63           adc     r15             
    61d4:       82 4e c2 15     mov     r14,    &0x15c2 
    61d8:       82 4f c4 15     mov     r15,    &0x15c4 
    61dc:       4f fb           and.b   r11,    r15     
    61de:       0b 16           popm.a  #1,     r11     
    61e0:       10 01           reta        
</pre>

For `TARGET=jn516x`
<pre>00083231 <tsch_random_byte>:
   83231:       8c c0 59 48 00 20               b.lwz      r6,0x4001298(r0)
   83237:       98 e0 b6 72 63 82               b.ori      r7,r0,0x41c64e6d
   8323d:       64 e6 3b                        b.mul      r7,r6,r7
   83240:       d8 e7 9c 0c                     b.addi     r7,r7,0x3039
   83244:       8c e0 19 48 00 20               b.sw       0x4001298(r0),r7
   8324a:       6c e7 09                        b.srli     r7,r7,0x10
   8324d:       60 63 38                        b.and      r3,r3,r7
   83250:       3e 03 18                        b.extbz    r3,r3
   83253:       47 d2 48                        b.jr       r9
</pre>

File: core/net/mac/tsch/tsch-log.h [hunk 1]
```diff
@@ -128,7 +128,7 @@ void tsch_log_process_pending(void);
     struct tsch_log_t *log = tsch_log_prepare_add(); \
     if(log != NULL) { \
       log->type = (log_type); \
-      init_code \
+      init_code; \
```
Comment (simonduq) [inline core/net/mac/tsch/tsch-log.h]: do we need some safe guards around it such parentheses?
(the problem was present before your fix but this occurs to me now)

**Comment by simonduq** [comment]
good to merge, let's wait for travis

**Comment by simonduq** [comment]
running out of space on z1 with security, maybe you could reduce the log level from 1 to 0 here:
https://github.com/EIT-ICT-RICH/contiki/blob/rich-3.x/examples/ipv6/rpl-tsch/project-conf.h#L151

**Comment by simonduq** [comment]
OK, my bad, the test starts with waiting for message confirming the association with security, but with log level 0 this never happens :/
Any clue how to solve this 2-byte overflow problem on z1? Looks like a linkerscript bug or something...

**Comment by simonduq** [comment]
you can also try to unset `UIP_CONF_ND6_SEND_NA` or `SICSLOWPAN_CONF_FRAG` (in the security-enabled case) to save space

**Comment by atiselsts** [comment]
Commited this, lets see what Travis says...
https://github.com/EIT-ICT-RICH/contiki/commit/34fb8ee3fbf129bb7aab2f888f0c6601ac370acd

Using 8-bit variables reduces RAM usage, reduces ROM usage, and
(subjectively) improves understandability of the code. I don't see problems
with using them instead of ints.

**Comment by simonduq** [comment]
I usually go for `int` whenever the size of the variable is irrelevant, because (I think) this is the fastest/most efficient type for the target mcu. I'm surprised `uint8_t` yields smaller ROM here! How much?

**Comment by simonduq** [comment]
I've now disabled NA on the z1 example, saving 2k ROM, so z1 shouldn't be a problem again for while!
Please just remove this commit and can merge https://github.com/EIT-ICT-RICH/contiki/commit/34fb8ee3fbf129bb7aab2f888f0c6601ac370acd

**Pull request merged**